diff --git a/src/swarm-doc/Swarm/Doc/Pedagogy.hs b/src/swarm-doc/Swarm/Doc/Pedagogy.hs
index d50807a..8ba006a 100644
--- a/src/swarm-doc/Swarm/Doc/Pedagogy.hs
+++ b/src/swarm-doc/Swarm/Doc/Pedagogy.hs
@@ -259,8 +259,8 @@ renderTutorialProgression =
     allUsed = concatMap mkTuplesForTutorial infos
 
     mkTuplesForTutorial tut =
-      map (\x -> (T.pack $ show x, tutIdxScenario)) $
-        M.keys $
-          novelSolutionCommands tut
+      map (\x -> (T.pack $ show x, tutIdxScenario))
+        $ M.keys
+        $ novelSolutionCommands tut
      where
       tutIdxScenario = tutInfo tut
diff --git a/src/swarm-doc/Swarm/Doc/Schema/Render.hs b/src/swarm-doc/Swarm/Doc/Schema/Render.hs
index 61e4416..784a704 100644
--- a/src/swarm-doc/Swarm/Doc/Schema/Render.hs
+++ b/src/swarm-doc/Swarm/Doc/Schema/Render.hs
@@ -84,9 +84,9 @@ makePandocTable titleMap (SchemaData _ (ToplevelSchema theTitle theDescription _
     ObjectProperties props -> makePropsTable True propertyColumnHeadings titleMap props
     ListMembers someStuff -> renderItems someStuff
     EnumMembers enumMembers ->
-      simpleTable [plain $ text "Member"] $
-        map (\m -> [plain $ code m]) $
-          NE.toList enumMembers
+      simpleTable [plain $ text "Member"]
+        $ map (\m -> [plain $ code m])
+        $ NE.toList enumMembers
 
 genPropsRow :: Bool -> Map SchemaIdReference T.Text -> (T.Text, SwarmSchema) -> [Blocks]
 genPropsRow includeDefaultColumn titleMap (k, x) =
@@ -122,9 +122,9 @@ genMarkdown schemaThings =
  where
   titleMap = makeTitleMap schemaThings
   pd =
-    mconcat $
-      map (makePandocTable titleMap) $
-        sortAndPruneSchemas (fromFilePath "scenario") schemaThings
+    mconcat
+      $ map (makePandocTable titleMap)
+      $ sortAndPruneSchemas (fromFilePath "scenario") schemaThings
 
 parseSchemaFile :: FileStemAndExtension -> IO (Either T.Text ToplevelSchema)
 parseSchemaFile stemAndExtension =
diff --git a/src/swarm-engine/Swarm/Game/Achievement/Description.hs b/src/swarm-engine/Swarm/Game/Achievement/Description.hs
index 0f1a7a2..7e17631 100644
--- a/src/swarm-engine/Swarm/Game/Achievement/Description.hs
+++ b/src/swarm-engine/Swarm/Game/Achievement/Description.hs
@@ -24,11 +24,11 @@ describe = \case
   GlobalAchievement CompletedAllTutorials ->
     AchievementInfo
       "Autodidact"
-      ( Just $
-          FTQuotation $
-            Quotation
-              "Terry Pratchet"
-              "I didn't go to university... But I have sympathy for those who did."
+      ( Just
+          $ FTQuotation
+          $ Quotation
+            "Terry Pratchet"
+            "I didn't go to university... But I have sympathy for those who did."
       )
       "Complete all of the tutorials."
       Moderate
@@ -80,11 +80,11 @@ describe = \case
   GameplayAchievement GetDisoriented ->
     AchievementInfo
       "Playing Ostrich"
-      ( Just $
-          FTQuotation $
-            Quotation
-              "Lil Jon"
-              "Fire up that loud / Another round of shots / Turn down for what?"
+      ( Just
+          $ FTQuotation
+          $ Quotation
+            "Lil Jon"
+            "Fire up that loud / Another round of shots / Turn down for what?"
       )
       "`turn down` without a compass. Congratulations, you are \"disoriented\". How are you supposed to move now?"
       Easy
diff --git a/src/swarm-engine/Swarm/Game/Scenario/Status.hs b/src/swarm-engine/Swarm/Game/Scenario/Status.hs
index 83ca95d..9240f99 100644
--- a/src/swarm-engine/Swarm/Game/Scenario/Status.hs
+++ b/src/swarm-engine/Swarm/Game/Scenario/Status.hs
@@ -145,15 +145,15 @@ updateScenarioInfoOnFinish
   completed
   si@(ScenarioInfo p prevPlayState) = case prevPlayState of
     Played launchParams (Metric _ (ProgressStats start _currentPlayMetrics)) prevBestRecords ->
-      ScenarioInfo p $
-        Played launchParams newPlayMetric $
-          updateBest newPlayMetric prevBestRecords
+      ScenarioInfo p
+        $ Played launchParams newPlayMetric
+        $ updateBest newPlayMetric prevBestRecords
      where
       el = (diffUTCTime `on` zonedTimeToUTC) z start
       cs = codeSizeFromDeterminator csd
       newCompletionFlag = if completed then Completed else Attempted
       newPlayMetric =
-        Metric newCompletionFlag $
-          ProgressStats start $
-            AttemptMetrics (DurationMetrics el ticks) cs
+        Metric newCompletionFlag
+          $ ProgressStats start
+          $ AttemptMetrics (DurationMetrics el ticks) cs
     _ -> si
diff --git a/src/swarm-engine/Swarm/Game/State/Robot.hs b/src/swarm-engine/Swarm/Game/State/Robot.hs
index c52b78e..a6992b5 100644
--- a/src/swarm-engine/Swarm/Game/State/Robot.hs
+++ b/src/swarm-engine/Swarm/Game/State/Robot.hs
@@ -224,9 +224,9 @@ wakeWatchingRobots myID currentTick loc = do
   let -- Step 1: Identify the robots that are watching this location.
       botsWatchingThisLoc :: [Robot]
       botsWatchingThisLoc =
-        mapMaybe (`IM.lookup` rMap) $
-          IS.toList $
-            MM.get loc watchingMap
+        mapMaybe (`IM.lookup` rMap)
+          $ IS.toList
+          $ MM.get loc watchingMap
 
       -- Step 2: Get the target wake time for each of these robots
       wakeTimes :: [(RID, TickNumber)]
diff --git a/src/swarm-engine/Swarm/Game/Step.hs b/src/swarm-engine/Swarm/Game/Step.hs
index 4618885..04dbe17 100644
--- a/src/swarm-engine/Swarm/Game/Step.hs
+++ b/src/swarm-engine/Swarm/Game/Step.hs
@@ -534,11 +534,11 @@ stepRobot r = do
   isCreative <- use creativeMode
   let shouldTrackActivity = isCreative || not (r' ^. systemRobot)
 
-  return $
-    applyWhen shouldTrackActivity (maintainActivityWindow t) $
-      r'
-        & machine .~ cesk'
-        & activityCounts . lifetimeStepCount +~ 1
+  return
+    $ applyWhen shouldTrackActivity (maintainActivityWindow t)
+    $ r'
+      & machine .~ cesk'
+      & activityCounts . lifetimeStepCount +~ 1
  where
   maintainActivityWindow t bot =
     bot & (activityCounts . activityWindow %~ WC.insert t)
diff --git a/src/swarm-engine/Swarm/Game/Step/Arithmetic.hs b/src/swarm-engine/Swarm/Game/Step/Arithmetic.hs
index 3c97a65..fc3c3af 100644
--- a/src/swarm-engine/Swarm/Game/Step/Arithmetic.hs
+++ b/src/swarm-engine/Swarm/Game/Step/Arithmetic.hs
@@ -82,9 +82,9 @@ incomparable = \case
 --   possible since the type system should catch it.
 incompatCmpErr :: Has (Throw Exn) sig m => Value -> Value -> m a
 incompatCmpErr v1 v2 =
-  throwError $
-    Fatal $
-      T.unwords ["Incompatible comparison of ", prettyValue v1, "and", prettyValue v2]
+  throwError
+    $ Fatal
+    $ T.unwords ["Incompatible comparison of ", prettyValue v1, "and", prettyValue v2]
 
 -- | Values were compared of a type which cannot be compared
 --   (e.g. functions, etc.).
diff --git a/src/swarm-engine/Swarm/Game/Step/Combustion.hs b/src/swarm-engine/Swarm/Game/Step/Combustion.hs
index b11e55c..72da06d 100644
--- a/src/swarm-engine/Swarm/Game/Step/Combustion.hs
+++ b/src/swarm-engine/Swarm/Game/Step/Combustion.hs
@@ -65,9 +65,9 @@ igniteCommand c d = do
   combustionDurationRand <- addCombustionBot e selfCombustibility createdAt loc
   let warmup = delay selfCombustibility
   let neighborAffectDuration = max 0 (combustionDurationRand - warmup)
-  when (neighborAffectDuration > 0) $
-    forM_ (getNeighborLocs loc) $
-      igniteNeighbor createdAt warmup neighborAffectDuration
+  when (neighborAffectDuration > 0)
+    $ forM_ (getNeighborLocs loc)
+    $ igniteNeighbor createdAt warmup neighborAffectDuration
  where
   verb = "ignite"
   verbed = "ignited"
diff --git a/src/swarm-engine/Swarm/Game/Step/Const.hs b/src/swarm-engine/Swarm/Game/Step/Const.hs
index 3306d3c..f4ae78b 100644
--- a/src/swarm-engine/Swarm/Game/Step/Const.hs
+++ b/src/swarm-engine/Swarm/Game/Step/Const.hs
@@ -175,17 +175,17 @@ execConst runChildProg c vs s k = do
       moveInDirection $ applyTurn (DRelative $ DPlanar DBack) $ orientation ? zero
     Volume -> case vs of
       [VInt limit] -> do
-        when (limit > globalMaxVolume) $
-          throwError $
-            CmdFailed
-              Volume
-              ( T.unwords
-                  [ "Can only measure up to"
-                  , T.pack $ show globalMaxVolume
-                  , "cells."
-                  ]
-              )
-              Nothing
+        when (limit > globalMaxVolume)
+          $ throwError
+          $ CmdFailed
+            Volume
+            ( T.unwords
+                [ "Can only measure up to"
+                , T.pack $ show globalMaxVolume
+                , "cells."
+                ]
+            )
+            Nothing
 
         robotLoc <- use robotLocation
         maybeResult <- floodFill robotLoc $ fromIntegral limit
@@ -206,9 +206,9 @@ execConst runChildProg c vs s k = do
               _ -> badConst
             else case goalVal of
               VPair (VInt x) (VInt y) ->
-                return $
-                  LocationTarget $
-                    Location (fromIntegral x) (fromIntegral y)
+                return
+                  $ LocationTarget
+                  $ Location (fromIntegral x) (fromIntegral y)
               _ -> badConst
         robotLoc <- use robotLocation
         result <- pathCommand $ PathfindingParameters maybeLimit robotLoc goal
@@ -244,17 +244,17 @@ execConst runChildProg c vs s k = do
       return $ mkReturn ()
     Stride -> case vs of
       [VInt d] -> do
-        when (d > fromIntegral maxStrideRange) $
-          throwError $
-            CmdFailed
-              Stride
-              ( T.unwords
-                  [ "Can only stride up to"
-                  , T.pack $ show maxStrideRange
-                  , "units."
-                  ]
-              )
-              Nothing
+        when (d > fromIntegral maxStrideRange)
+          $ throwError
+          $ CmdFailed
+            Stride
+            ( T.unwords
+                [ "Can only stride up to"
+                , T.pack $ show maxStrideRange
+                , "units."
+                ]
+            )
+            Nothing
 
         -- Figure out where we're going
         loc <- use robotLocation
@@ -264,9 +264,9 @@ execConst runChildProg c vs s k = do
         -- Excludes the base location.
         let locsInDirection :: [Cosmic Location]
             locsInDirection =
-              take (min (fromIntegral d) maxStrideRange) $
-                drop 1 $
-                  iterate (`offsetBy` heading) loc
+              take (min (fromIntegral d) maxStrideRange)
+                $ drop 1
+                $ iterate (`offsetBy` heading) loc
 
         failureMaybes <- mapM checkMoveFailure locsInDirection
         let maybeFirstFailure = asum failureMaybes
@@ -423,9 +423,9 @@ execConst runChildProg c vs s k = do
           robotInventory %= delete item
 
           -- Log a special message if re-equipping life support system
-          when (itemName == "life support system") $
-            void $
-              traceLog Logged Warning "Life support system back online! Life failure averted!"
+          when (itemName == "life support system")
+            $ void
+            $ traceLog Logged Warning "Life support system back online! Life failure averted!"
 
           -- Check whether we should bestow the 'EquippedAllDevices' achievement
           curScenario <- use currentScenarioPath
@@ -1385,9 +1385,9 @@ execConst runChildProg c vs s k = do
     -- add the targeted entity so it can be consumed by the recipe
     let makeRecipe r = (,r) <$> make' (insert nextE inv, ins) r
     chosenRecipe <-
-      weightedChoice (\((_, _), r) -> r ^. recipeWeight) $
-        rights $
-          map makeRecipe recipes
+      weightedChoice (\((_, _), r) -> r ^. recipeWeight)
+        $ rights
+        $ map makeRecipe recipes
     ((invTaken, outs), recipe) <-
       chosenRecipe
         `isJustOrFail` ["You don't have the ingredients to", verbPhrase, indefinite (nextE ^. entityName) <> "."]
diff --git a/src/swarm-engine/Swarm/Game/Step/Path/Cache.hs b/src/swarm-engine/Swarm/Game/Step/Path/Cache.hs
index c1b8e8d..2dc41bf 100644
--- a/src/swarm-engine/Swarm/Game/Step/Path/Cache.hs
+++ b/src/swarm-engine/Swarm/Game/Step/Path/Cache.hs
@@ -84,19 +84,19 @@ retrieveCachedPath currentWalkabilityContext newParms = do
         PathfindingParameters prevDistLimit previousSubworldName t = prevParms
 
     -- Subworlds must match
-    unless (previousSubworldName == currentSubworld) $
-      Left $
-        DifferentArg NewSubworld
+    unless (previousSubworldName == currentSubworld)
+      $ Left
+      $ DifferentArg NewSubworld
 
     -- Pathfinding target type must match
-    unless (t == target) $
-      Left $
-        DifferentArg NewTargetType
+    unless (t == target)
+      $ Left
+      $ DifferentArg NewTargetType
 
     -- Walkability context must match
-    unless (currentWalkabilityContext == prevWalkabilityContext) $
-      Left $
-        DifferentArg NewWalkabilityContext
+    unless (currentWalkabilityContext == prevWalkabilityContext)
+      $ Left
+      $ DifferentArg NewWalkabilityContext
 
     left (DifferentArg . NewDistanceLimit) $
       getDistanceLimitInvalidation currentRobotLoc pathCells currentDistanceLimit prevDistLimit
diff --git a/src/swarm-lang/Swarm/Language/Capability.hs b/src/swarm-lang/Swarm/Language/Capability.hs
index ebbcbc0..562a380 100644
--- a/src/swarm-lang/Swarm/Language/Capability.hs
+++ b/src/swarm-lang/Swarm/Language/Capability.hs
@@ -149,6 +149,6 @@ constCaps = \case
 -- | Inverts the 'constCaps' mapping.
 constByCaps :: Map Capability (NE.NonEmpty Const)
 constByCaps =
-  binTuples $
-    map swap $
-      mapMaybe (sequenceA . (id &&& constCaps)) allConst
+  binTuples
+    $ map swap
+    $ mapMaybe (sequenceA . (id &&& constCaps)) allConst
diff --git a/src/swarm-lang/Swarm/Language/LSP.hs b/src/swarm-lang/Swarm/Language/LSP.hs
index d558b04..e48362d 100644
--- a/src/swarm-lang/Swarm/Language/LSP.hs
+++ b/src/swarm-lang/Swarm/Language/LSP.hs
@@ -39,30 +39,30 @@ import Witch
 
 lspMain :: IO ()
 lspMain =
-  void $
-    runServer $
-      ServerDefinition
-        { defaultConfig = ()
-        , configSection = "swarm"
-        , parseConfig = const $ const $ Right ()
-        , onConfigChange = const $ return ()
-        , doInitialize = \env _req -> pure $ Right env
-        , staticHandlers = const handlers
-        , interpretHandler = \env -> Iso (runLspT env) liftIO
-        , options =
-            defaultOptions
-              { -- set sync options to get DidSave event, as well as Open and Close events.
-                optTextDocumentSync =
-                  Just
-                    ( LSP.TextDocumentSyncOptions
-                        (Just True)
-                        (Just syncKind)
-                        (Just False)
-                        (Just False)
-                        (Just . LSP.InR . LSP.SaveOptions $ Just True)
-                    )
-              }
-        }
+  void
+    $ runServer
+    $ ServerDefinition
+      { defaultConfig = ()
+      , configSection = "swarm"
+      , parseConfig = const $ const $ Right ()
+      , onConfigChange = const $ return ()
+      , doInitialize = \env _req -> pure $ Right env
+      , staticHandlers = const handlers
+      , interpretHandler = \env -> Iso (runLspT env) liftIO
+      , options =
+          defaultOptions
+            { -- set sync options to get DidSave event, as well as Open and Close events.
+              optTextDocumentSync =
+                Just
+                  ( LSP.TextDocumentSyncOptions
+                      (Just True)
+                      (Just syncKind)
+                      (Just False)
+                      (Just False)
+                      (Just . LSP.InR . LSP.SaveOptions $ Just True)
+                  )
+            }
+      }
  where
   -- Using SyncFull seems to handle the debounce for us.
   -- The alternative is to use SyncIncremental, but then the
diff --git a/src/swarm-lang/Swarm/Language/LSP/Hover.hs b/src/swarm-lang/Swarm/Language/LSP/Hover.hs
index 75dd902..7caccff 100644
--- a/src/swarm-lang/Swarm/Language/LSP/Hover.hs
+++ b/src/swarm-lang/Swarm/Language/LSP/Hover.hs
@@ -96,13 +96,13 @@ explain trm = case trm ^. sTerm of
   -- definition or bindings
   SLet ls isRecursive var mTypeAnn _ _ rhs _b -> pure $ explainDefinition ls isRecursive var (rhs ^. sType) mTypeAnn
   SLam (Loc _s v) _mType _syn ->
-    pure $
-      typeSignature v ty $
-        "A lambda expression binding the variable " <> U.bquote v <> "."
+    pure
+      $ typeSignature v ty
+      $ "A lambda expression binding the variable " <> U.bquote v <> "."
   SBind mv _ _ _ rhs _cmds ->
-    pure $
-      typeSignature (maybe "__rhs" locVal mv) (getInnerType $ rhs ^. sType) $
-        "A monadic bind for commands" <> maybe "." (\(Loc _s v) -> ", that binds variable " <> U.bquote v <> ".") mv
+    pure
+      $ typeSignature (maybe "__rhs" locVal mv) (getInnerType $ rhs ^. sType)
+      $ "A monadic bind for commands" <> maybe "." (\(Loc _s v) -> ", that binds variable " <> U.bquote v <> ".") mv
   -- composite types
   SPair {} ->
     Node
diff --git a/src/swarm-lang/Swarm/Language/Syntax/Pattern.hs b/src/swarm-lang/Swarm/Language/Syntax/Pattern.hs
index fdc4d8a..c6ebf77 100644
--- a/src/swarm-lang/Swarm/Language/Syntax/Pattern.hs
+++ b/src/swarm-lang/Swarm/Language/Syntax/Pattern.hs
@@ -61,8 +61,8 @@ type UTerm = Term' UType
 -- | Raw parsed syntax, without comments or type annotations.
 pattern Syntax :: SrcLoc -> Term -> Syntax
 pattern Syntax l t <- Syntax' l t _ ()
-  where
-    Syntax l t = Syntax' l t Empty ()
+ where
+  Syntax l t = Syntax' l t Empty ()
 
 {-# COMPLETE Syntax #-}
 
@@ -81,8 +81,8 @@ noLoc = Syntax mempty
 pattern STerm :: Term -> Syntax
 pattern STerm t <-
   CSyntax _ t _
-  where
-    STerm t = Syntax mempty t
+ where
+  STerm t = Syntax mempty t
 
 pattern TRequirements :: Text -> Term -> Term
 pattern TRequirements x t = SRequirements x (STerm t)
@@ -94,8 +94,8 @@ pattern TPair t1 t2 = SPair (STerm t1) (STerm t2)
 -- | Match a TLam without annotations.
 pattern TLam :: Var -> Maybe Type -> Term -> Term
 pattern TLam v ty t <- SLam (locVal -> v) ty (STerm t)
-  where
-    TLam v ty t = SLam (Loc NoLoc v) ty (STerm t)
+ where
+  TLam v ty t = SLam (Loc NoLoc v) ty (STerm t)
 
 -- | Match a TApp without annotations.
 pattern TApp :: Term -> Term -> Term
@@ -110,20 +110,20 @@ pattern (:$:) t1 s2 = SApp (STerm t1) s2
 -- | Match a TLet without annotations.
 pattern TLet :: LetSyntax -> Bool -> Var -> Maybe RawPolytype -> Maybe Polytype -> Maybe Requirements -> Term -> Term -> Term
 pattern TLet ls r v mty mpty mreq t1 t2 <- SLet ls r (locVal -> v) mty mpty mreq (STerm t1) (STerm t2)
-  where
-    TLet ls r v mty mpty mreq t1 t2 = SLet ls r (Loc NoLoc v) mty mpty mreq (STerm t1) (STerm t2)
+ where
+  TLet ls r v mty mpty mreq t1 t2 = SLet ls r (Loc NoLoc v) mty mpty mreq (STerm t1) (STerm t2)
 
 -- | Match a STydef without annotations.
 pattern TTydef :: TDVar -> Polytype -> Maybe TydefInfo -> Term -> Term
 pattern TTydef v ty mtd t1 <- STydef (locVal -> v) ty mtd (STerm t1)
-  where
-    TTydef v ty mtd t1 = STydef (Loc NoLoc v) ty mtd (STerm t1)
+ where
+  TTydef v ty mtd t1 = STydef (Loc NoLoc v) ty mtd (STerm t1)
 
 -- | Match a TBind without annotations.
 pattern TBind :: Maybe Var -> Maybe Polytype -> Maybe Requirements -> Term -> Term -> Term
 pattern TBind mv mty mreq t1 t2 <- SBind (fmap locVal -> mv) _ mty mreq (STerm t1) (STerm t2)
-  where
-    TBind mv mty mreq t1 t2 = SBind (Loc NoLoc <$> mv) Nothing mty mreq (STerm t1) (STerm t2)
+ where
+  TBind mv mty mreq t1 t2 = SBind (Loc NoLoc <$> mv) Nothing mty mreq (STerm t1) (STerm t2)
 
 -- | Match a TDelay without annotations.
 pattern TDelay :: Term -> Term
@@ -132,8 +132,8 @@ pattern TDelay t = SDelay (STerm t)
 -- | Match a TRcd without annotations.
 pattern TRcd :: [(LocVar, Maybe Term)] -> Term
 pattern TRcd m <- SRcd ((map . second . fmap) _sTerm -> m)
-  where
-    TRcd m = SRcd ((map . second . fmap) STerm m)
+ where
+  TRcd m = SRcd ((map . second . fmap) STerm m)
 
 pattern TProj :: Term -> Var -> Term
 pattern TProj t x = SProj (STerm t) x
diff --git a/src/swarm-lang/Swarm/Language/Typecheck.hs b/src/swarm-lang/Swarm/Language/Typecheck.hs
index de5768b..e4e6dc9 100644
--- a/src/swarm-lang/Swarm/Language/Typecheck.hs
+++ b/src/swarm-lang/Swarm/Language/Typecheck.hs
@@ -1264,9 +1264,9 @@ check s@(CSyntax l t cs) expected = addLocToTypeErr l $ case t of
     -- Now check the type of the body, under a context extended with
     -- the type and requirements of the bound variable.
     t2' <-
-      withBinding (locVal x) upty $
-        withBinding (locVal x) reqs $
-          check t2 expected
+      withBinding (locVal x) upty
+        $ withBinding (locVal x) reqs
+        $ check t2 expected
 
     -- Make sure none of the generated skolem variables have escaped.
     ask @UCtx >>= mapM_ (noSkolems l skolems)
@@ -1324,9 +1324,9 @@ check s@(CSyntax l t cs) expected = addLocToTypeErr l $ case t of
         let fieldMap = M.fromList $ map (first locVal) fields
             expectedFields = M.keysSet tyMap
             actualFields = M.keysSet fieldMap
-        when (actualFields /= expectedFields) $
-          throwTypeErr l $
-            FieldsMismatch (joined expectedFields actualFields)
+        when (actualFields /= expectedFields)
+          $ throwTypeErr l
+          $ FieldsMismatch (joined expectedFields actualFields)
         -- Since we checked above that 'fields' and 'tyMap' have the
         -- same keys, we know this lookup into the tyMap will never fail;
         -- however, we still use lookup + mapMaybe to avoid partial functions.
diff --git a/src/swarm-lang/Swarm/Language/Types.hs b/src/swarm-lang/Swarm/Language/Types.hs
index 386b573..510a42a 100644
--- a/src/swarm-lang/Swarm/Language/Types.hs
+++ b/src/swarm-lang/Swarm/Language/Types.hs
@@ -654,8 +654,8 @@ pattern TyBase b = TyConApp (TCBase b) []
 
 pattern TyVar :: Var -> Type
 pattern TyVar x <- Fix (TyVarF _ x)
-  where
-    TyVar x = Fix (TyVarF x x)
+ where
+  TyVar x = Fix (TyVarF x x)
 
 pattern TyVoid :: Type
 pattern TyVoid = TyBase BVoid
@@ -725,8 +725,8 @@ pattern UTyBase b = UTyConApp (TCBase b) []
 
 pattern UTyVar :: Var -> UType
 pattern UTyVar x <- Free (TyVarF _ x)
-  where
-    UTyVar x = Free (TyVarF x x)
+ where
+  UTyVar x = Free (TyVarF x x)
 
 pattern UTyVar' :: Var -> Var -> UType
 pattern UTyVar' x y = Free (TyVarF x y)
diff --git a/src/swarm-scenario/Swarm/Game/Scenario.hs b/src/swarm-scenario/Swarm/Game/Scenario.hs
index accb311..d42dfea 100644
--- a/src/swarm-scenario/Swarm/Game/Scenario.hs
+++ b/src/swarm-scenario/Swarm/Game/Scenario.hs
@@ -323,9 +323,9 @@ instance FromJSONE ScenarioInputs Scenario where
             ]
 
       let mergedWaypoints =
-            M.fromList $
-              map (worldName &&& runIdentity . waypoints . navigation) $
-                NE.toList allWorlds
+            M.fromList
+              $ map (worldName &&& runIdentity . waypoints . navigation)
+              $ NE.toList allWorlds
 
       mergedPortals <-
         validatePortals
diff --git a/src/swarm-scenario/Swarm/Game/Scenario/Objective/Graph.hs b/src/swarm-scenario/Swarm/Game/Scenario/Objective/Graph.hs
index 2d13c96..535c96a 100644
--- a/src/swarm-scenario/Swarm/Game/Scenario/Objective/Graph.hs
+++ b/src/swarm-scenario/Swarm/Game/Scenario/Objective/Graph.hs
@@ -83,9 +83,9 @@ getNegatedIds objs =
 
 getObjectivesById :: [Objective] -> Map ObjectiveLabel Objective
 getObjectivesById objs =
-  M.fromList $
-    map swap $
-      mapMaybe (sequenceA . (id &&& view objectiveId)) objs
+  M.fromList
+    $ map swap
+    $ mapMaybe (sequenceA . (id &&& view objectiveId)) objs
 
 -- | Uses the textual labels for those objectives that
 -- have them, and assigns arbitrary integer IDs for
diff --git a/src/swarm-scenario/Swarm/Game/Scenario/Topography/WorldPalette.hs b/src/swarm-scenario/Swarm/Game/Scenario/Topography/WorldPalette.hs
index 180db89..d019aed 100644
--- a/src/swarm-scenario/Swarm/Game/Scenario/Topography/WorldPalette.hs
+++ b/src/swarm-scenario/Swarm/Game/Scenario/Topography/WorldPalette.hs
@@ -112,9 +112,9 @@ prepForJson (PaletteAndMaskChar (StructurePalette _ _ suggestedPalette) maybeMas
 
   unassignedCells :: M.Map (TerrainWith EntityName) (TerrainWith EntityFacade)
   unassignedCells =
-    M.withoutKeys entityCells $
-      Set.fromList $
-        map (toKey . snd) preassignments
+    M.withoutKeys entityCells
+      $ Set.fromList
+      $ map (toKey . snd) preassignments
 
   (maskCharacter, availableCharacterPool) = case maybeMaskChar of
     Just c -> (c, genericCharacterPool)
diff --git a/src/swarm-scenario/Swarm/Game/State/Landscape.hs b/src/swarm-scenario/Swarm/Game/State/Landscape.hs
index 627f6b8..07caa7b 100644
--- a/src/swarm-scenario/Swarm/Game/State/Landscape.hs
+++ b/src/swarm-scenario/Swarm/Game/State/Landscape.hs
@@ -143,9 +143,9 @@ buildWorld tem WorldDescription {..} =
   (robots worldName, first getTerrainIndex . wf)
  where
   getTerrainIndex t =
-    M.findWithDefault 0 t $
-      terrainIndexByName $
-        tem ^. terrainMap
+    M.findWithDefault 0 t
+      $ terrainIndexByName
+      $ tem ^. terrainMap
 
   g = gridContent area
 
diff --git a/src/swarm-scenario/Swarm/Game/World/DSL/Load.hs b/src/swarm-scenario/Swarm/Game/World/DSL/Load.hs
index d8685a8..4b128b0 100644
--- a/src/swarm-scenario/Swarm/Game/World/DSL/Load.hs
+++ b/src/swarm-scenario/Swarm/Game/World/DSL/Load.hs
@@ -46,7 +46,7 @@ loadWorld tem (fp, src) = do
     liftEither . left (AssetNotLoaded (Data Worlds) fp . SystemFailure . CanNotParseMegaparsec) $
       runParser parseWExp src
   t <-
-    withThrow (AssetNotLoaded (Data Worlds) fp . SystemFailure . DoesNotTypecheck . prettyText @CheckErr) $
-      runReader tem . runReader @WorldMap M.empty $
-        infer CNil wexp
+    withThrow (AssetNotLoaded (Data Worlds) fp . SystemFailure . DoesNotTypecheck . prettyText @CheckErr)
+      $ runReader tem . runReader @WorldMap M.empty
+      $ infer CNil wexp
   return (into @Text (dropExtension (takeFileName fp)), t)
diff --git a/src/swarm-topography/Swarm/Game/Scenario/Topography/Palette.hs b/src/swarm-topography/Swarm/Game/Scenario/Topography/Palette.hs
index 55f5ca1..b752f35 100644
--- a/src/swarm-topography/Swarm/Game/Scenario/Topography/Palette.hs
+++ b/src/swarm-topography/Swarm/Game/Scenario/Topography/Palette.hs
@@ -56,12 +56,12 @@ instance (FromJSONE e a) => FromJSONE e (StructurePalette a) where
     verifyChar = traverse $ ensureSingleChar . K.toString
     ensureSingleChar [x] = return x
     ensureSingleChar x =
-      fail $
-        T.unpack $
-          T.unwords
-            [ "Palette entry is not a single character:"
-            , quote $ T.pack x
-            ]
+      fail
+        $ T.unpack
+        $ T.unwords
+          [ "Palette entry is not a single character:"
+          , quote $ T.pack x
+          ]
 
 ------------------------------------------------------------
 -- Parsing character sets
diff --git a/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure.hs b/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure.hs
index 0ebf065..cd77338 100644
--- a/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure.hs
+++ b/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure.hs
@@ -102,21 +102,21 @@ paintMap ::
 paintMap maskChar pal g = do
   nestedLists <- mapM toCell g
   forM_ maskChar $ \c ->
-    unless (Set.notMember c paletteKeys) $
-      fail $
-        unwords
-          [ "Mask character"
-          , ['"', c, '"']
-          , "overlaps palette entry"
-          ]
-
-  unless (Set.null unusedPaletteChars) $
-    fail $
-      unwords
-        [ "Unused characters in palette:"
-        , intercalate ", " $ map pure $ Set.toList unusedPaletteChars
+    unless (Set.notMember c paletteKeys)
+      $ fail
+      $ unwords
+        [ "Mask character"
+        , ['"', c, '"']
+        , "overlaps palette entry"
         ]
 
+  unless (Set.null unusedPaletteChars)
+    $ fail
+    $ unwords
+      [ "Unused characters in palette:"
+      , intercalate ", " $ map pure $ Set.toList unusedPaletteChars
+      ]
+
   let cells = fmap standardCell <$> nestedLists
       wps = catMaybes $ mapWithCoords getWp nestedLists
 
diff --git a/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Assembly.hs b/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Assembly.hs
index aae2a21..948689f 100644
--- a/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Assembly.hs
+++ b/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Assembly.hs
@@ -251,27 +251,27 @@ validatePlacement ns (Pose _ orientation) = do
       n = getStructureName . name $ ns
 
   when (isRecognizable ns) $ do
-    when (flipped orientation) $
-      Left $
-        T.unwords
-          [ "Placing recognizable structure"
-          , quote n
-          , "with flipped orientation is not supported."
-          ]
+    when (flipped orientation)
+      $ Left
+      $ T.unwords
+        [ "Placing recognizable structure"
+        , quote n
+        , "with flipped orientation is not supported."
+        ]
 
     -- Redundant orientations by rotational symmetry are accounted
     -- for at scenario parse time
-    when (Set.notMember placementDirection recognizedOrientations) $
-      Left $
-        T.unwords
-          [ "Placing recognizable structure"
-          , quote n
-          , "with"
-          , renderDir placementDirection
-          , "orientation is not supported."
-          , "Try"
-          , commaList $ map renderDir $ Set.toList recognizedOrientations
-          , "instead."
-          ]
+    when (Set.notMember placementDirection recognizedOrientations)
+      $ Left
+      $ T.unwords
+        [ "Placing recognizable structure"
+        , quote n
+        , "with"
+        , renderDir placementDirection
+        , "orientation is not supported."
+        , "Try"
+        , commaList $ map renderDir $ Set.toList recognizedOrientations
+        , "instead."
+        ]
  where
   renderDir = quote . T.pack . directionJsonModifier . show
diff --git a/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Precompute.hs b/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Precompute.hs
index a9f6210..3a4947b 100644
--- a/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Precompute.hs
+++ b/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Precompute.hs
@@ -113,9 +113,9 @@ extractOrientedGrid ::
   AbsoluteDir ->
   StructureWithGrid b a
 extractOrientedGrid (ExtractedArea x neGrid) d =
-  StructureWithGrid d w $
-    ExtractedArea x $
-      applyOrientationTransformNE (Orientation d False) neGrid
+  StructureWithGrid d w
+    $ ExtractedArea x
+    $ applyOrientationTransformNE (Orientation d False) neGrid
  where
   w = RowWidth . rectWidth . getNEGridDimensions $ neGrid
 
@@ -146,9 +146,9 @@ lookupStaticPlacements (StaticStructureInfo theAutomatons thePlacements) =
    where
     g (LocatedStructure (OrientedStructure theName d) loc) = do
       sGrid <- M.lookup theName definitionMap
-      return $
-        PositionedStructure (Cosmic subworldName loc) $
-          extractOrientedGrid (grid $ annotatedGrid sGrid) d
+      return
+        $ PositionedStructure (Cosmic subworldName loc)
+        $ extractOrientedGrid (grid $ annotatedGrid sGrid) d
 
 -- | Matches definitions against the placements.
 -- Fails fast (short-circuits) if a non-matching
diff --git a/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Prep.hs b/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Prep.hs
index e24d765..0ad2e64 100644
--- a/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Prep.hs
+++ b/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Prep.hs
@@ -93,9 +93,9 @@ mkEntityLookup grids =
     -- We wrap the entities with 'Just' since the Aho-Corasick
     -- matcher needs to compare against world cells, which are of 'Maybe' type.
     chunksStateMachine =
-      makeStateMachine $
-        map (NE.toList . fmap Just &&& id) $
-          HS.toList extractedChunksForStateMachine
+      makeStateMachine
+        $ map (NE.toList . fmap Just &&& id)
+        $ HS.toList extractedChunksForStateMachine
 
   -- The values of this map are guaranteed to contain only one
   -- entry per row of each structure, even if some of those
@@ -141,9 +141,9 @@ explodeRowEntities annotatedRow@(ConsolidatedRowReferences rowMembers _ width) =
   chunks = getContiguousChunks $ NE.toList rowMembers
 
   f (e, occurrences) =
-    SingleRowEntityOccurrences annotatedRow e chunks $
-      sconcat $
-        NE.map deriveEntityOffsets occurrences
+    SingleRowEntityOccurrences annotatedRow e chunks
+      $ sconcat
+      $ NE.map deriveEntityOffsets occurrences
 
   -- Tuples of (entity, rowOccurrenceOfEntity).
   -- Only row members for which an entity exists (is not Nothing)
diff --git a/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Registry.hs b/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Registry.hs
index 9f6a207..0d37731 100644
--- a/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Registry.hs
+++ b/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Registry.hs
@@ -117,9 +117,9 @@ populateStaticFoundStructures allFound =
   byLocation = M.unions $ map mkOccupationMap resolvedCollisions
 
   byName =
-    M.map (NEM.fromList . NE.map ((upperLeftCorner &&& rotatedTo . structureWithGrid) &&& structureWithGrid)) $
-      binTuples $
-        map (name . originalItem . entityGrid . structureWithGrid &&& id) resolvedCollisions
+    M.map (NEM.fromList . NE.map ((upperLeftCorner &&& rotatedTo . structureWithGrid) &&& structureWithGrid))
+      $ binTuples
+      $ map (name . originalItem . entityGrid . structureWithGrid &&& id) resolvedCollisions
 
   resolvePreplacementCollisions foundList =
     nonOverlappingFound <> maybeToList (listToMaybe overlapsByDecreasingPreference)
diff --git a/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Tracking.hs b/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Tracking.hs
index 52929c0..2aabec9 100644
--- a/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Tracking.hs
+++ b/src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Tracking.hs
@@ -181,14 +181,14 @@ checkChunksCombination
     candidateExpected = concatMap NE.toList candidateExpectedLists
 
     foundRowChunksLookup =
-      fmap NEIS.fromList $
-        binTuplesHM $
-          map (pVal &&& pIndex) candidatesChunked
+      fmap NEIS.fromList
+        $ binTuplesHM
+        $ map (pVal &&& pIndex) candidatesChunked
 
     (candidateFailures, candidateExpectedLists) =
-      partitionEithers $
-        map (checkCandidateAgainstObservedChunks horizontalOffsets foundRowChunksLookup) $
-          NE.toList rowChunkReferences
+      partitionEithers
+        $ map (checkCandidateAgainstObservedChunks horizontalOffsets foundRowChunksLookup)
+        $ NE.toList rowChunkReferences
 
     mkFoundStructures x =
       NE.toList $ NE.map mkFoundStructure . referencingRows . chunkStructure $ foundChunkRow x
@@ -210,9 +210,9 @@ checkCandidateAgainstObservedChunks ::
   Either (ChunkMatchFailureReason e) (NE.NonEmpty (ChunkedRowMatch (ConsolidatedRowReferences b e) e))
 checkCandidateAgainstObservedChunks horizontalOffsets foundRowChunksLookup (RowChunkMatchingReference r chunkPositionMap) =
   left (ChunkMatchFailureReason $ renderSharedNames r) $ do
-    unless isKeysSubset . Left $
-      NoKeysSubset $
-        (FoundChunkComparison `on` HS.toList) foundChunksKeys referenceChunksKeys
+    unless isKeysSubset . Left
+      $ NoKeysSubset
+      $ (FoundChunkComparison `on` HS.toList) foundChunksKeys referenceChunksKeys
 
     nonEmptyPairs <-
       maybeToEither EmptyIntersection $
@@ -354,9 +354,9 @@ seedPossibleOffsets (sizeDifference, FoundAndExpectedChunkPositions found expect
   NEIS.nonEmptySet $ IS.fromList possibleOffsets
  where
   possibleOffsets =
-    NE.take (sizeDifference + 1) $
-      NE.map (subtract (NEIS.findMin expected)) $
-        NEIS.toAscList found
+    NE.take (sizeDifference + 1)
+      $ NE.map (subtract (NEIS.findMin expected))
+      $ NEIS.toAscList found
 
 -- | Return all of the offsets that are viable for repetitions of this chunk.
 --
diff --git a/src/swarm-tournament/Swarm/Web/Tournament.hs b/src/swarm-tournament/Swarm/Web/Tournament.hs
index 86019d1..5a04b06 100644
--- a/src/swarm-tournament/Swarm/Web/Tournament.hs
+++ b/src/swarm-tournament/Swarm/Web/Tournament.hs
@@ -327,9 +327,9 @@ makeCookieHeader val =
 
 doLogout :: Maybe TL.Text -> LoginHandler
 doLogout maybeRefererUrl =
-  return $
-    addHeader (fromMaybe defaultRedirectPage maybeRefererUrl) $
-      addHeader ((makeCookieHeader "") {setCookieMaxAge = Just 0}) NoContent
+  return
+    $ addHeader (fromMaybe defaultRedirectPage maybeRefererUrl)
+    $ addHeader ((makeCookieHeader "") {setCookieMaxAge = Just 0}) NoContent
 
 doLoginResponse ::
   AuthenticationStorage IO ->
@@ -339,9 +339,9 @@ doLoginResponse ::
 doLoginResponse authStorage refererUrl userAlias = do
   cookieString <-
     liftIO $ cookieFromUsername authStorage userAlias
-  return $
-    addHeader refererUrl $
-      addHeader (makeCookieHeader $ LBS.toStrict $ encodeUtf8 cookieString) NoContent
+  return
+    $ addHeader refererUrl
+    $ addHeader (makeCookieHeader $ LBS.toStrict $ encodeUtf8 cookieString) NoContent
 
 -- * Web app declaration
 
diff --git a/src/swarm-tournament/Swarm/Web/Tournament/Validate.hs b/src/swarm-tournament/Swarm/Web/Tournament/Validate.hs
index a089a57..68dc2a6 100644
--- a/src/swarm-tournament/Swarm/Web/Tournament/Validate.hs
+++ b/src/swarm-tournament/Swarm/Web/Tournament/Validate.hs
@@ -68,9 +68,9 @@ validateScenarioUpload (CommonValidationArgs solnTimeout persistenceArgs) gameVe
  where
   computeMetrics file = do
     (gs, scenarioObject) <-
-      withExceptT ScenarioUploadInstantiationFailure $
-        gamestateFromScenarioText $
-          fileContent file
+      withExceptT ScenarioUploadInstantiationFailure
+        $ gamestateFromScenarioText
+        $ fileContent file
 
     soln <- except $ maybeToEither NoSolutionProvided $ gs ^. winSolution
 
@@ -137,9 +137,9 @@ validateSubmittedSolution (CommonValidationArgs solnTimeout persistenceArgs) sce
                 <$> scenarioLookupFunc scenarioSha1
             )
 
-      fmap fst $
-        withExceptT RetrievedInstantiationFailure $
-          gamestateFromScenarioText scenarioContent
+      fmap fst
+        $ withExceptT RetrievedInstantiationFailure
+        $ gamestateFromScenarioText scenarioContent
 
     solnMetrics <-
       withExceptT SubmittedSolutionEvaluationFailure $
@@ -170,9 +170,9 @@ initScenarioObject ::
   ExceptT ScenarioInstantiationFailure IO Scenario
 initScenarioObject scenarioInputs content = do
   rawYaml <- withExceptT YamlDecodeError . except . decodeEither' $ LBS.toStrict content
-  withExceptT ScenarioParseFailure $
-    except $
-      parseEither (parseJSONE' scenarioInputs) rawYaml
+  withExceptT ScenarioParseFailure
+    $ except
+    $ parseEither (parseJSONE' scenarioInputs) rawYaml
 
 gamestateFromScenarioText ::
   LBS.ByteString ->
diff --git a/src/swarm-tui/Swarm/TUI/Controller.hs b/src/swarm-tui/Swarm/TUI/Controller.hs
index 6b85e1b..32c8c5f 100644
--- a/src/swarm-tui/Swarm/TUI/Controller.hs
+++ b/src/swarm-tui/Swarm/TUI/Controller.hs
@@ -242,9 +242,9 @@ handleMainMenuEvent menu = \case
         uiState . uiMenu .= MessagesMenu
       About -> do
         uiState . uiMenu .= AboutMenu
-        Brick.zoom (playState . progression) $
-          attainAchievement $
-            GlobalAchievement LookedAtAboutScreen
+        Brick.zoom (playState . progression)
+          $ attainAchievement
+          $ GlobalAchievement LookedAtAboutScreen
       Quit -> haltApp
   CharKey 'q' -> haltApp
   ControlChar 'q' -> haltApp
diff --git a/src/swarm-tui/Swarm/TUI/Controller/EventHandlers/Robot.hs b/src/swarm-tui/Swarm/TUI/Controller/EventHandlers/Robot.hs
index bec2010..017e0ca 100644
--- a/src/swarm-tui/Swarm/TUI/Controller/EventHandlers/Robot.hs
+++ b/src/swarm-tui/Swarm/TUI/Controller/EventHandlers/Robot.hs
@@ -46,9 +46,9 @@ handleRobotPanelEvent bev = do
     Nothing -> case bev of
       VtyEvent ev@(V.EvKey k m) -> do
         handled <- handleKey keyHandler k m
-        unless handled $
-          Brick.zoom (playState . scenarioState . uiGameplay) $
-            handleInventoryListEvent ev
+        unless handled
+          $ Brick.zoom (playState . scenarioState . uiGameplay)
+          $ handleInventoryListEvent ev
       _ -> continueWithoutRedraw
 
 -- | Handle key events in the robot panel.
diff --git a/src/swarm-tui/Swarm/TUI/Controller/SaveScenario.hs b/src/swarm-tui/Swarm/TUI/Controller/SaveScenario.hs
index b106839..41d40a2 100644
--- a/src/swarm-tui/Swarm/TUI/Controller/SaveScenario.hs
+++ b/src/swarm-tui/Swarm/TUI/Controller/SaveScenario.hs
@@ -45,9 +45,9 @@ applyCompletionAchievements ::
   EventM n ProgressionState ()
 applyCompletionAchievements won t p = do
   forM_ (listToMaybe $ splitDirectories p) $ \firstDir -> do
-    when (won && firstDir == tutorialsDirname) $
-      attainAchievement' t (Just $ ScenarioPath p) $
-        GlobalAchievement CompletedSingleTutorial
+    when (won && firstDir == tutorialsDirname)
+      $ attainAchievement' t (Just $ ScenarioPath p)
+      $ GlobalAchievement CompletedSingleTutorial
 
   -- Check if all tutorials have been completed
   tutorialMap <- use $ scenarios . to getTutorials . to scMap
@@ -57,9 +57,9 @@ applyCompletionAchievements won t p = do
       -- like the right thing to do
       isComplete (SICollection _ (SC m)) = all isComplete m
 
-  when (all isComplete tutorialMap) $
-    attainAchievement $
-      GlobalAchievement CompletedAllTutorials
+  when (all isComplete tutorialMap)
+    $ attainAchievement
+    $ GlobalAchievement CompletedAllTutorials
 
 saveScenarioInfoOnFinish ::
   FilePath ->
diff --git a/src/swarm-tui/Swarm/TUI/Controller/UpdateUI.hs b/src/swarm-tui/Swarm/TUI/Controller/UpdateUI.hs
index f5eb1c4..e3f0ae7 100644
--- a/src/swarm-tui/Swarm/TUI/Controller/UpdateUI.hs
+++ b/src/swarm-tui/Swarm/TUI/Controller/UpdateUI.hs
@@ -241,9 +241,9 @@ updateUI = do
 
   -- Update the robots modal only when it is enabled.  See #2370.
   curModal <- use $ playState . scenarioState . uiGameplay . uiDialogs . uiModal
-  when ((view modalType <$> curModal) == Just (MidScenarioModal RobotsModal)) $
-    Brick.zoom (playState . scenarioState . uiGameplay . uiDialogs . uiRobot) $
-      doRobotListUpdate dOps g
+  when ((view modalType <$> curModal) == Just (MidScenarioModal RobotsModal))
+    $ Brick.zoom (playState . scenarioState . uiGameplay . uiDialogs . uiRobot)
+    $ doRobotListUpdate dOps g
 
   let shouldRedraw =
         worldPanelUpdated
diff --git a/src/swarm-tui/Swarm/TUI/Editor/View.hs b/src/swarm-tui/Swarm/TUI/Editor/View.hs
index fc7c265..c4d4462 100644
--- a/src/swarm-tui/Swarm/TUI/Editor/View.hs
+++ b/src/swarm-tui/Swarm/TUI/Editor/View.hs
@@ -62,9 +62,9 @@ drawWorldEditor toplevelFocusRing uig =
         ]
 
   innerWidget =
-    padLeftRight 1 $
-      hLimit 30 $
-        controlsBox <=> statusBox
+    padLeftRight 1
+      $ hLimit 30
+      $ controlsBox <=> statusBox
 
   worldEditor = uig ^. uiWorldEditor
   maybeAreaBounds = worldEditor ^. editingBounds . boundsRect
@@ -118,13 +118,13 @@ drawWorldEditor toplevelFocusRing uig =
         ]
 
   renderBounds (upperLeftCoord, lowerRightCoord) =
-    str $
-      unwords $
-        L.intersperse
-          "@"
-          [ EA.renderRectDimensions rectArea
-          , locationToString upperLeftLoc
-          ]
+    str
+      $ unwords
+      $ L.intersperse
+        "@"
+        [ EA.renderRectDimensions rectArea
+        , locationToString upperLeftLoc
+        ]
    where
     upperLeftLoc = coordsToLoc upperLeftCoord
     lowerRightLoc = coordsToLoc lowerRightCoord
diff --git a/src/swarm-tui/Swarm/TUI/Launch/View.hs b/src/swarm-tui/Swarm/TUI/Launch/View.hs
index c033c18..8c12a42 100644
--- a/src/swarm-tui/Swarm/TUI/Launch/View.hs
+++ b/src/swarm-tui/Swarm/TUI/Launch/View.hs
@@ -34,9 +34,9 @@ drawFileBrowser b =
   centerLayer $ hLimit 50 $ ui <=> help
  where
   ui =
-    vLimit 15 $
-      borderWithLabel (txt "Choose a file") $
-        FB.renderFileBrowser True b
+    vLimit 15
+      $ borderWithLabel (txt "Choose a file")
+      $ FB.renderFileBrowser True b
 
   footerRows =
     map
@@ -48,9 +48,9 @@ drawFileBrowser b =
       ]
 
   help =
-    padTop (Pad 1) $
-      vBox $
-        [ case FB.fileBrowserException b of
+    padTop (Pad 1)
+      $ vBox
+      $ [ case FB.fileBrowserException b of
             Nothing -> emptyWidget
             Just e ->
               hCenter
@@ -59,7 +59,7 @@ drawFileBrowser b =
                 . T.pack
                 $ E.displayException e
         ]
-          <> footerRows
+        <> footerRows
 
 optionDescription :: ScenarioConfigPanelFocusable -> Maybe Text
 optionDescription = \case
@@ -94,9 +94,9 @@ drawLaunchConfigPanel (LaunchOptions lc launchParams) =
       $ txt label
 
   mkSeedEditorWidget =
-    hLimit 10 $
-      overrideAttr E.editFocusedAttr customEditFocusedAttr $
-        renderEditor (txt . mconcat) (isFocused SeedSelector) seedEditor
+    hLimit 10
+      $ overrideAttr E.editFocusedAttr customEditFocusedAttr
+      $ renderEditor (txt . mconcat) (isFocused SeedSelector) seedEditor
   seedEntryWidget = case seedVal launchParams of
     Left _ -> mkSeedEditorWidget
     Right x -> mkSeedEntryWidget x
@@ -111,12 +111,12 @@ drawLaunchConfigPanel (LaunchOptions lc launchParams) =
       else case seedEntryContent of
         Just x -> str $ show x
         Nothing ->
-          withDefAttr dimAttr $
-            txt $
-              T.unwords
-                [ "scenario default"
-                , parens $ T.pack scenarioSeedText
-                ]
+          withDefAttr dimAttr
+            $ txt
+            $ T.unwords
+              [ "scenario default"
+              , parens $ T.pack scenarioSeedText
+              ]
 
   unspecifiedFileMessage =
     if isFocused ScriptSelector
@@ -171,12 +171,12 @@ drawLaunchConfigPanel (LaunchOptions lc launchParams) =
         $ infoContent
 
     padControl widgetName label widgetObj =
-      padBottom (Pad 1) $
-        padLeft (Pad 2) $
-          hBox
-            [ mkButton widgetName (label <> ": ")
-            , widgetObj
-            ]
+      padBottom (Pad 1)
+        $ padLeft (Pad 2)
+        $ hBox
+          [ mkButton widgetName (label <> ": ")
+          , widgetObj
+          ]
 
     controlsBox =
       vBox
diff --git a/src/swarm-tui/Swarm/TUI/Model/StateUpdate.hs b/src/swarm-tui/Swarm/TUI/Model/StateUpdate.hs
index 06fc5b5..485bbbf 100644
--- a/src/swarm-tui/Swarm/TUI/Model/StateUpdate.hs
+++ b/src/swarm-tui/Swarm/TUI/Model/StateUpdate.hs
@@ -441,9 +441,9 @@ setUIGameplay gs curTime isAutoplaying siPair@(ScenarioWith scenario _) uig =
   entityList = EU.getEntitiesForList $ gs ^. landscape . terrainAndEntities . entityMap
 
   (isEmptyArea, newBounds) =
-    EU.getEditingBounds $
-      NE.head $
-        scenario ^. scenarioLandscape . scenarioWorlds
+    EU.getEditingBounds
+      $ NE.head
+      $ scenario ^. scenarioLandscape . scenarioWorlds
 
   setNewBounds maybeOldBounds =
     if isEmptyArea
diff --git a/src/swarm-tui/Swarm/TUI/View.hs b/src/swarm-tui/Swarm/TUI/View.hs
index b1e25e0..0b6efd4 100644
--- a/src/swarm-tui/Swarm/TUI/View.hs
+++ b/src/swarm-tui/Swarm/TUI/View.hs
@@ -273,9 +273,9 @@ drawNewGameMenuUI appState (l :| ls) launchOptions = case displayedFor of
   drawDescription (SISingle (ScenarioWith s (ScenarioPath sPath))) =
     vBox
       [ drawMarkdown (nonBlank (s ^. scenarioOperation . scenarioDescription))
-      , cached (ScenarioPreview sPath) $
-          hCenter . padTop (Pad 1) . vLimit 6 $
-            hLimitPercent 60 worldPeek
+      , cached (ScenarioPreview sPath)
+          $ hCenter . padTop (Pad 1) . vLimit 6
+          $ hLimitPercent 60 worldPeek
       , padTop (Pad 1) table
       ]
    where
@@ -285,9 +285,9 @@ drawNewGameMenuUI appState (l :| ls) launchOptions = case displayedFor of
 
     worldTuples = buildWorldTuples $ s ^. scenarioLandscape
     theWorlds =
-      genMultiWorld worldTuples $
-        fromMaybe 0 $
-          s ^. scenarioLandscape . scenarioSeed
+      genMultiWorld worldTuples
+        $ fromMaybe 0
+        $ s ^. scenarioLandscape . scenarioSeed
 
     entIsKnown =
       getEntityIsKnown $
@@ -318,9 +318,9 @@ drawNewGameMenuUI appState (l :| ls) launchOptions = case displayedFor of
     padTopLeft = padTop (Pad 1) . padLeft (Pad 1)
 
     tableRows =
-      map (map padTopLeft . pairToList) $
-        mapMaybe sequenceA $
-          firstRow : secondRow : maybe [] (makeBestScoreRows . view scenarioStatus) currentScenarioInfo
+      map (map padTopLeft . pairToList)
+        $ mapMaybe sequenceA
+        $ firstRow : secondRow : maybe [] (makeBestScoreRows . view scenarioStatus) currentScenarioInfo
     table =
       BT.renderTable
         . BT.surroundingBorder False
@@ -352,22 +352,22 @@ describeProgress (Metric p (ProgressStats _startedAt (AttemptMetrics (DurationMe
         <> maybeToList (sizeDisplay <$> maybeCodeMetrics)
    where
     sizeDisplay (ScenarioCodeMetrics myCharCount myAstSize) =
-      withAttr greenAttr $
-        vBox $
-          map
-            txt
-            [ T.unwords
-                [ "Code:"
-                , T.pack $ show myCharCount
-                , "chars"
+      withAttr greenAttr
+        $ vBox
+        $ map
+          txt
+          [ T.unwords
+              [ "Code:"
+              , T.pack $ show myCharCount
+              , "chars"
+              ]
+          , (" " <>)
+              $ parens
+              $ T.unwords
+                [ T.pack $ show myAstSize
+                , "AST nodes"
                 ]
-            , (" " <>) $
-                parens $
-                  T.unwords
-                    [ T.pack $ show myAstSize
-                    , "AST nodes"
-                    ]
-            ]
+          ]
  where
   formatTimeDiff :: NominalDiffTime -> Text
   formatTimeDiff = T.pack . formatTime defaultTimeLocale "%hh %Mm %Ss"
@@ -395,11 +395,11 @@ makeBestScoreRows scenarioStat =
     hasMultiple = length groups > 1
 
   makeBestRow hasDistinctByCriteria (b, criteria) =
-    ( hLimit (maxLeftColumnWidth + 2) $
-        vBox $
-          [ padLeft Max $ txt "best:"
+    ( hLimit (maxLeftColumnWidth + 2)
+        $ vBox
+        $ [ padLeft Max $ txt "best:"
           ]
-            <> elaboratedCriteria
+          <> elaboratedCriteria
     , Just $ describeProgress b
     )
    where
@@ -671,11 +671,11 @@ drawModal h ps isNoMenu = \case
     EntityPaletteModal -> EV.drawEntityPaintSelector uig
   EndScenarioModal x -> case x of
     ScenarioFinishModal outcome ->
-      padBottom (Pad 1) $
-        vBox $
-          map
-            (hCenter . txt)
-            content
+      padBottom (Pad 1)
+        $ vBox
+        $ map
+          (hCenter . txt)
+          content
      where
       content = case outcome of
         WinModal -> ["Congratulations!"]
@@ -685,10 +685,10 @@ drawModal h ps isNoMenu = \case
           ]
     QuitModal -> padBottom (Pad 1) $ hCenter $ txt (quitMsg isNoMenu)
     KeepPlayingModal ->
-      padLeftRight 1 $
-        displayParagraphs $
-          pure
-            "Have fun!  Hit Ctrl-Q whenever you're ready to proceed to the next challenge or return to the menu."
+      padLeftRight 1
+        $ displayParagraphs
+        $ pure
+          "Have fun!  Hit Ctrl-Q whenever you're ready to proceed to the next challenge or return to the menu."
  where
   gs = ps ^. gameState
   uig = ps ^. uiGameplay
@@ -820,17 +820,17 @@ commandsListWidget aMap gs =
   base = gs ^? baseRobot
   entsByCap = case base of
     Just r ->
-      M.map NE.toList $
-        entitiesByCapability $
-          (r ^. equippedDevices) `union` (r ^. robotInventory)
+      M.map NE.toList
+        $ entitiesByCapability
+        $ (r ^. equippedDevices) `union` (r ^. robotInventory)
     Nothing -> mempty
 
   listDevices cmd = vBox $ map (drawLabelledEntityName aMap) providerDevices
    where
     providerDevices =
-      concatMap (flip (M.findWithDefault []) entsByCap) $
-        maybeToList $
-          constCaps cmd
+      concatMap (flip (M.findWithDefault []) entsByCap)
+        $ maybeToList
+        $ constCaps cmd
 
 -- | Generate a pop-up widget to display the description of an entity.
 descriptionWidget :: ScenarioState -> Entity -> Widget Name
diff --git a/src/swarm-tui/Swarm/TUI/View/Objective.hs b/src/swarm-tui/Swarm/TUI/View/Objective.hs
index 8d0224c..4cd99c0 100644
--- a/src/swarm-tui/Swarm/TUI/View/Objective.hs
+++ b/src/swarm-tui/Swarm/TUI/View/Objective.hs
@@ -56,14 +56,14 @@ renderGoalsDisplay gd desc =
   lw = _listWidget gd
   fr = _focus gd
   leftSide =
-    hLimitPercent 30 $
-      padAll 1 $
-        vBox
-          [ hCenter $ str "Goals"
-          , padAll 1 $
-              vLimit 10 $
-                withFocusRing fr (BL.renderList drawGoalListItem) lw
-          ]
+    hLimitPercent 30
+      $ padAll 1
+      $ vBox
+        [ hCenter $ str "Goals"
+        , padAll 1
+            $ vLimit 10
+            $ withFocusRing fr (BL.renderList drawGoalListItem) lw
+        ]
 
   -- Adds very subtle coloring to indicate focus switch
   highlightIfFocused = case (hasMultiple, focusGetCurrent fr) of
diff --git a/src/swarm-tui/Swarm/TUI/View/Robot/Details.hs b/src/swarm-tui/Swarm/TUI/View/Robot/Details.hs
index 0488264..3cd853b 100644
--- a/src/swarm-tui/Swarm/TUI/View/Robot/Details.hs
+++ b/src/swarm-tui/Swarm/TUI/View/Robot/Details.hs
@@ -69,6 +69,6 @@ renderRobotDetails r paneState =
     ]
 
   commandHistogramEntries =
-    map mkHistogramEntry $
-      M.toList $
-        r ^. activityCounts . commandsHistogram
+    map mkHistogramEntry
+      $ M.toList
+      $ r ^. activityCounts . commandsHistogram
diff --git a/src/swarm-tui/Swarm/TUI/View/Structure.hs b/src/swarm-tui/Swarm/TUI/View/Structure.hs
index 3f27fa2..babeafb 100644
--- a/src/swarm-tui/Swarm/TUI/View/Structure.hs
+++ b/src/swarm-tui/Swarm/TUI/View/Structure.hs
@@ -142,14 +142,14 @@ renderStructuresDisplay aMap gs structureDisplay =
   lw = _structurePanelListWidget structureDisplay
   fr = _structurePanelFocus structureDisplay
   leftSide =
-    hLimitPercent 25 $
-      padAll 1 $
-        vBox
-          [ hCenter $ withAttr boldAttr $ txt "Candidates"
-          , padAll 1 $
-              vLimit 10 $
-                withFocusRing fr (BL.renderList drawSidebarListItem) lw
-          ]
+    hLimitPercent 25
+      $ padAll 1
+      $ vBox
+        [ hCenter $ withAttr boldAttr $ txt "Candidates"
+        , padAll 1
+            $ vLimit 10
+            $ withFocusRing fr (BL.renderList drawSidebarListItem) lw
+        ]
 
   -- Adds very subtle coloring to indicate focus switch
   highlightIfFocused = case focusGetCurrent fr of
diff --git a/src/swarm-web/Swarm/Web.hs b/src/swarm-web/Swarm/Web.hs
index 84ed802..019a3cf 100644
--- a/src/swarm-web/Swarm/Web.hs
+++ b/src/swarm-web/Swarm/Web.hs
@@ -248,9 +248,9 @@ navigationHandler appStateRef = do
 recogLogHandler :: IO AppState -> Handler [SearchLog EntityName]
 recogLogHandler appStateRef = do
   appState <- liftIO appStateRef
-  return $
-    map (fmap (view entityName)) $
-      appState ^. playState . scenarioState . gameState . discovery . structureRecognition . recognitionLog
+  return
+    $ map (fmap (view entityName))
+    $ appState ^. playState . scenarioState . gameState . discovery . structureRecognition . recognitionLog
 
 recogFoundHandler :: IO AppState -> Handler [StructureLocation]
 recogFoundHandler appStateRef = do
diff --git a/test/bench/Benchmark.hs b/test/bench/Benchmark.hs
index 1712bd7..dbaba56 100644
--- a/test/bench/Benchmark.hs
+++ b/test/bench/Benchmark.hs
@@ -197,12 +197,12 @@ main = do
     newLabel =
       unwords
         [ label
-        , T.unpack $
-            parens $
-              T.unwords
-                [ showT ticks
-                , "ticks"
-                ]
+        , T.unpack
+            $ parens
+            $ T.unwords
+              [ showT ticks
+              , "ticks"
+              ]
         ]
 
   robotNumbers = [10, 20 .. 40]
diff --git a/test/integration/Main.hs b/test/integration/Main.hs
index 2df5e0b..693a44b 100644
--- a/test/integration/Main.hs
+++ b/test/integration/Main.hs
@@ -135,9 +135,9 @@ testNoLoadingErrors r =
 checkNoRuntimeErrors :: RuntimeState -> IO ()
 checkNoRuntimeErrors r =
   forM_ (r ^. eventLog . notificationsContent) $ \e ->
-    when (isError e) $
-      assertFailure $
-        show (e ^. leSeverity) <> " was produced during loading: " <> T.unpack (e ^. leText)
+    when (isError e)
+      $ assertFailure
+      $ show (e ^. leSeverity) <> " was produced during loading: " <> T.unpack (e ^. leText)
 
 isError :: LogEntry -> Bool
 isError = (>= Warning) . view leSeverity
diff --git a/test/unit/TestPretty.hs b/test/unit/TestPretty.hs
index b29c888..26ed41a 100644
--- a/test/unit/TestPretty.hs
+++ b/test/unit/TestPretty.hs
@@ -40,9 +40,9 @@ testPrettyConst =
         )
     , testCase
         "operators #8 - double unary negation"
-        ( equalPrettyLine "-(-1)" $
-            TApp (TConst Neg) $
-              TApp (TConst Neg) (TInt 1)
+        ( equalPrettyLine "-(-1)"
+            $ TApp (TConst Neg)
+            $ TApp (TConst Neg) (TInt 1)
         )
     , testCase
         "operators #8 - unary negation with strongly fixing binary operator"
@@ -194,36 +194,36 @@ testPrettyConst =
         ]
     , testGroup
         "tydef"
-        [ testCase "tydef alias" $
-            equalPrettyLine "tydef X = Int end" $
-              TTydef "X" (mkTrivPoly TyInt) Nothing (TConst Noop)
-        , testCase "tydef Maybe" $
-            equalPrettyLine "tydef Maybe a = Unit + a end" $
-              TTydef "Maybe" (mkQPoly (TyUnit :+: TyVar "a")) Nothing (TConst Noop)
-        , testCase "tydef multi-arg" $
-            equalPrettyLine "tydef Foo a b c d = Unit + ((a * b) + ((c -> d) * a)) end" $
-              TTydef
-                "Foo"
-                ( mkQPoly
-                    (TyUnit :+: (TyVar "a" :*: TyVar "b") :+: ((TyVar "c" :->: TyVar "d") :*: TyVar "a"))
-                )
-                Nothing
-                (TConst Noop)
-        , testCase "consecutive tydef" $
-            equalPrettyLine "tydef X = Int end\n\ntydef Y = Bool end" $
-              TTydef "X" (mkTrivPoly TyInt) Nothing (TTydef "Y" (mkTrivPoly TyBool) Nothing (TConst Noop))
+        [ testCase "tydef alias"
+            $ equalPrettyLine "tydef X = Int end"
+            $ TTydef "X" (mkTrivPoly TyInt) Nothing (TConst Noop)
+        , testCase "tydef Maybe"
+            $ equalPrettyLine "tydef Maybe a = Unit + a end"
+            $ TTydef "Maybe" (mkQPoly (TyUnit :+: TyVar "a")) Nothing (TConst Noop)
+        , testCase "tydef multi-arg"
+            $ equalPrettyLine "tydef Foo a b c d = Unit + ((a * b) + ((c -> d) * a)) end"
+            $ TTydef
+              "Foo"
+              ( mkQPoly
+                  (TyUnit :+: (TyVar "a" :*: TyVar "b") :+: ((TyVar "c" :->: TyVar "d") :*: TyVar "a"))
+              )
+              Nothing
+              (TConst Noop)
+        , testCase "consecutive tydef"
+            $ equalPrettyLine "tydef X = Int end\n\ntydef Y = Bool end"
+            $ TTydef "X" (mkTrivPoly TyInt) Nothing (TTydef "Y" (mkTrivPoly TyBool) Nothing (TConst Noop))
         ]
     , testGroup
         "recursive types"
-        [ testCase "nat" $
-            equalPrettyLine "rec n. Unit + n" $
-              TyRec "n" (TyUnit :+: Fix (TyRecVarF NZ))
-        , testCase "list" $
-            equalPrettyLine "rec list. Unit + (a * list)" $
-              TyRec "list" (TyUnit :+: (TyVar "a" :*: Fix (TyRecVarF NZ)))
-        , testCase "rose" $
-            equalPrettyLine "rec r. a * (rec l. Unit + (r * l))" $
-              TyRec "r" (TyVar "a" :*: TyRec "l" (TyUnit :+: (Fix (TyRecVarF (NS NZ)) :*: Fix (TyRecVarF NZ))))
+        [ testCase "nat"
+            $ equalPrettyLine "rec n. Unit + n"
+            $ TyRec "n" (TyUnit :+: Fix (TyRecVarF NZ))
+        , testCase "list"
+            $ equalPrettyLine "rec list. Unit + (a * list)"
+            $ TyRec "list" (TyUnit :+: (TyVar "a" :*: Fix (TyRecVarF NZ)))
+        , testCase "rose"
+            $ equalPrettyLine "rec r. a * (rec l. Unit + (r * l))"
+            $ TyRec "r" (TyVar "a" :*: TyRec "l" (TyUnit :+: (Fix (TyRecVarF (NS NZ)) :*: Fix (TyRecVarF NZ))))
         ]
     ]
  where
diff --git a/test/unit/TestScoring.hs b/test/unit/TestScoring.hs
index 2efed8a..adb0821 100644
--- a/test/unit/TestScoring.hs
+++ b/test/unit/TestScoring.hs
@@ -74,18 +74,18 @@ betterReplTimeAfterCodeSizeRecord =
  where
   mkZonedTime t = ZonedTime (LocalTime (fromOrdinalDate 2023 1) (TimeOfDay 0 t 0)) utc
   newRunWithoutCodeSize =
-    Metric Completed $
-      ProgressStats (mkZonedTime 1) $
-        AttemptMetrics
-          (DurationMetrics 1 $ TickNumber 1)
-          Nothing
+    Metric Completed
+      $ ProgressStats (mkZonedTime 1)
+      $ AttemptMetrics
+        (DurationMetrics 1 $ TickNumber 1)
+        Nothing
 
   oldCompletedRunWithCodeSize =
-    Metric Completed $
-      ProgressStats (mkZonedTime 0) $
-        AttemptMetrics
-          (DurationMetrics 2 $ TickNumber 2)
-          (Just $ ScenarioCodeMetrics 1 1)
+    Metric Completed
+      $ ProgressStats (mkZonedTime 0)
+      $ AttemptMetrics
+        (DurationMetrics 2 $ TickNumber 2)
+        (Just $ ScenarioCodeMetrics 1 1)
 
   oldBestWithCodeSize =
     BestRecords
@@ -114,18 +114,18 @@ betterCodeWorseTime =
  where
   mkZonedTime t = ZonedTime (LocalTime (fromOrdinalDate 2023 1) (TimeOfDay 0 t 0)) utc
   newRunBetterCodeSize =
-    Metric Completed $
-      ProgressStats (mkZonedTime 1) $
-        AttemptMetrics
-          (DurationMetrics 2 $ TickNumber 2)
-          (Just $ ScenarioCodeMetrics 1 1)
+    Metric Completed
+      $ ProgressStats (mkZonedTime 1)
+      $ AttemptMetrics
+        (DurationMetrics 2 $ TickNumber 2)
+        (Just $ ScenarioCodeMetrics 1 1)
 
   oldRunPoorCodeSize =
-    Metric Completed $
-      ProgressStats (mkZonedTime 0) $
-        AttemptMetrics
-          (DurationMetrics 1 $ TickNumber 1)
-          (Just $ ScenarioCodeMetrics 2 2)
+    Metric Completed
+      $ ProgressStats (mkZonedTime 0)
+      $ AttemptMetrics
+        (DurationMetrics 1 $ TickNumber 1)
+        (Just $ ScenarioCodeMetrics 2 2)
 
   oldBests =
     BestRecords
